### PR TITLE
Upgrade Tags system

### LINKED ISSUE #12: Tags endpoint to list all tags dynamically

### LINKED ISSUE #13: Tags frontend in List
Show tags in list of frontend (retrieve these from backend built in #12)

Before:
image

After:
image

### LINKED ISSUE #14: Update filterlist dynamically based on #12
Update filterlist dynamically based on #12 

image

### PR DESCRIPTION
**Pull request by svemat01**
Use a dedicated dynamic tag system

Tags button in the admin menu
`/admin`

Tags menu
`/admin/tags`

Edit tag
`/admin/tags/KEY/edit`

New tag
`/admin/tags/new`

Tags show up as display name

Registered as key in post, keys not in tags list don't show up *(test is the key for "Hello3" and luc is a non existant tag)*

Closes #12
Closes #13
Closes #14

### CODE REVIEW

**Comment by svemat01** [comment]
Closes #12 #13 #14

File: backend/src/routes/tags/list.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { redis } from '../..';
+
+// const HARDCODED_precompute: string[] = [];
+
+// const HARDCODED_list = Object.assign(
+//     {},
+//     ...HARDCODED_precompute.map((value) => ({
+//         [value.toLowerCase().replace(/[^'a-z]/g, '')]: value,
+//     }))
+// );
```
Comment (lucemans) [inline backend/src/routes/tags/list.ts:12]:
```suggestion
```

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -57,10 +59,16 @@ export const App = () => {
                                     path="/admin/:id/edit"
                                     element={<AdminPostEdit />}
                                 />
+                                <Route path="/admin/tags" element={<Tags />} />
                                 <Route
-                                    path="/grant/:id"
-                                    element={<Grant />}
+                                    path="/admin/tags/new"
+                                    element={<AdminTagEdit isNew />}
```
Comment (lucemans) [inline frontend/src/App.tsx:65]: ?

**Review by lucemans — APPROVED** [review]
LGTM

**Review by lucemans — CHANGES_REQUESTED** [review]
Pending changes...

**Review by lucemans — APPROVED** [review]
LGTM

**Pull request merged**